### PR TITLE
Mob 1325

### PR DESCRIPTION
**Pull request by muryno**
## **[Ticket MOB-1325](https://fattmerchant.atlassian.net/browse/MOB-1325)**

> **Shipping amount and po number not showing in the payments or invoices pages**

## What did I do?

Fix bug on iOS SDK 2.2.1 | Shipping amount and po number not showing in the payments or invoices pages

---

## Screenshot(s):

* [Paste_screenshots_here]

---

## PR notes contain:

* [x] Checklist for any deployment dependencies (such as new env vars, back-end dependencies, etc)

## PR Checklist:

* [ ] **I tested this**
* [x] I updated from the base branch (usually `latest`)
* [x] I attached screenshot(s) (if there were UI changes)
* [x] I created `data-testid` attributes for any new UI
* [ ] I used the following PR title format: "[SPRINT-XX][fat-xx] BLAH"
  * Add "[EFS]" to the PR title if it was in the ticket title, e.g. "[SPRINT-XX][fat-xx][EFS]"
* [ ] I looked over the Files Changed tab
* [x] I consulted with the ticket's product owner (if necessary)
* [x] I requested a peer review

### CODE REVIEW

File: fattmerchant-ios-sdk/Cardpresent/Usecase/TakePayment.swift [hunk 1]
```diff
@@ -99,11 +99,11 @@ class TakePayment {
   ///   - amount: the Amount to charge
   ///   - paymentMethodId: the id of the PaymentMethod to charge
   /// - Returns: a ChargeRequest
-  internal static func createChargeRequest(amount: Amount, paymentMethodId: String) -> ChargeRequest {
+    internal static func createChargeRequest(amount: Amount, paymentMethodId: String, poNumber: String?  ) -> ChargeRequest {
```
Comment (omar-salah-elboredy) [inline fattmerchant-ios-sdk/Cardpresent/Usecase/TakePayment.swift:103]: Kindly update the documentation comment for `poNumber`

File: fattmerchant_ios_sdkTests/Usecase/TakePaymentTests.swift [hunk 2]
```diff
@@ -32,11 +32,13 @@ class TakePaymentTests: XCTestCase {
 
   func testCreateChargeRequest() {
     let amount = Amount(dollars: 10.00)
+    let poNumber = "asdf4terstges"
```
Comment (omar-salah-elboredy) [inline fattmerchant_ios_sdkTests/Usecase/TakePaymentTests.swift:35]: I think we also need to add another test case which includes empty/null poNumber to check how the behaviour would go.

**Review by omar-salah-elboredy — CHANGES_REQUESTED** [review]
Kindly address these quick changes

**Comment by omar-salah-elboredy** [comment]
Please also add screenshots or a video showing what has been done in this PR/Ticket, specially for bug fixes, for easier review and to be as a proof of your fix that it worked as expected.

File: fattmerchant_ios_sdkTests/Usecase/TakePaymentTests.swift (hunk 2 above)
Comment (muryno) [reply fattmerchant_ios_sdkTests/Usecase/TakePaymentTests.swift:35]: > Please also add screenshots or a video showing what has been done in this PR/Ticket, specially for bug fixes, for easier review and to be as a proof of your fix that it worked as expected.

I don't have card reader to be able to get picture of the transaction.. rather I can test with ensuring the test case work fine...

**Comment by thomasvidas** [comment]
@muryno I added the template for this PR 😄 I'll make sure going forward that PRs on our repos have right template!

File: fattmerchant-ios-sdk/Cardpresent/Usecase/TakePayment.swift [hunk 3]
```diff
@@ -94,16 +94,17 @@ class TakePayment {
...
   ///   - amount: the Amount to charge
   ///   - paymentMethodId: the id of the PaymentMethod to charge
+    /// - poNumber: The purchase order number for the transaction
   /// - Returns: a ChargeRequest
-  internal static func createChargeRequest(amount: Amount, paymentMethodId: String) -> ChargeRequest {
+    internal static func createChargeRequest(amount: Amount, paymentMethodId: String, poNumber: String?  ) -> ChargeRequest {
     let chargeRequestMeta = [
-      "subtotal": amount.dollarsString()
+      "subtotal": amount.dollarsString(),
+      "poNumber": poNumber ?? ""
```
Comment (thomasvidas) [inline fattmerchant-ios-sdk/Cardpresent/Usecase/TakePayment.swift]: This isn't the correct use case, if there is no `poNumber` passed in, don't set anything, not even an empty string. It should be something like this

```
let chargeRequestMeta = ["subtotal": amount.dollarsString()]

if poNumber != null
  chargeRequestMeta.poNumber = poNumber
```

**Review by thomasvidas — CHANGES_REQUESTED** [review]
Besides the one change, this seems fine

**Review by thomasvidas — APPROVED** [review]
